### ISSUE TITLE
Threads support (blocked on Threads)

### ISSUE THREAD

**Issue by enn-nafnlaus** [snarfed/bridgy-fed#1210]
A month ago we saw the first account bridged from Threads to ATProto:

https://bsky.app/profile/shnarfed.threads.net.ap.brid.gy

However, it seems to just be a test case at this point. I cannot find any other bridged accounts.  I asked a friend on Threads to see if he could follow @bsky.brid.gy@bsky.brid.gy on Threads, as per the docs (https://fed.brid.gy/docs).  He cannot find the account.  The docs state that Threads is supported, yet provides no other information than that.

What is the actual state?  If it's not complete, is a complete implementation coming?  Thanks!

**Comment by snarfed** [comment]
Thanks for filing! Good to have an issue to track this one.

Threads's fediverse support is very incomplete right now. You can share out to the fediverse, and now you can see some amount of replies, likes, and reposts from the fediverse, but you can't see or interact with original fediverse posts, or find or follow people on the fediverse. That includes the `@bsky.brid.gy@bsky.brid.gy` bot user, which is the immediate technical reason that Threads users can't yet opt into the bridge.

The bigger point, though, is that bridging is complicated and confusing enough right now for people on "normal" fediverse instances. I'm not ready to do anything special to support Threads as long as its fediverse integration is so partial and incomplete.

I'm excited to support it as soon as they're fully ready though! Theoretically BF won't have to do anything new, it will "just work." Supposedly. We'll see!

**Comment by snarfed** [comment]
Oh, and you're right, the docs shouldn't say that Threads is supported. I'll fix that. Thanks for the nudge!

**Comment by enn-nafnlaus** [comment]
Got it.  Hopefully it comes sooner or later!  I was excited to find that an old Twitter friend was now active on threads, thinking we could just bridge... guess not... for the time being.  :(

Thanks for your work!  :)

**Comment by enn-nafnlaus** [comment]
> Oh, and you're right, the docs shouldn't say that Threads is supported. I'll fix that. Thanks for the nudge!

While you're fixing the docs:

_"To follow a web site, first [enter it here](https://fed.brid.gy/docs#enter-web-site) to make sure it's set up, then wait a minute, then [search for it in Bluesky](https://bsky.app/search)"_

The "enter it here" link does nothing.

**Comment by mackuba** [comment]
Just wondering, if Threads adds the ability to follow Fediverse accounts at some point, will it be possible to opt-in to bridging from there then without any changes, or is that somehow blocked explicitly on your side at the moment?

**Comment by snarfed** [comment]
Good question. I'm not explicitly blocking Threads at all right now. I am a bit concerned that when they let Threads users follow fediverse users, other parts of federation may still be missing or incomplete, which will cause trouble. Guess we'll see!

**Comment by antont** [comment]
I'm also curious. You mentioned earlier that somehow Threads' AP implementation is broken / incomplete so that bridging wouldn't (really) work? I wonder if it would help anything to know the details and maybe try to make Meta fix some of the problems. I guess not, they do what they do following their own roadmap and strategy.

**Comment by TomCasavant** [comment]
> I'm also curious. You mentioned earlier that somehow Threads' AP implementation is broken / incomplete so that bridging wouldn't (really) work? I wonder if it would help anything to know the details and maybe try to make Meta fix some of the problems. I guess not, they do what they do following their own roadmap and strategy.

In order to opt in your account to the bridge you need to be able to follow the bridgy account, Threads' implementation is incomplete because that's not possible yet

There could be potential issues with their implementation *after* they let you follow fediverse accounts, but we won't know until they roll out that feature

**Comment by snarfed** [comment]
Lots more recent Threads discussion in https://github.com/snarfed/bridgy-fed/issues/1471#issuecomment-2474092931 and https://github.com/snarfed/bridgy-fed/issues/1305#issuecomment-2474451008 and below.

**Comment by TomCasavant** [comment]
> In order to opt in your account to the bridge you need to be able to follow the bridgy account, Threads' implementation is incomplete because that's not possible yet


FWIW that is possible now, [I tested it](https://tomkahe.com/@tom/113595990018344861). The bridge creates the bridged account, https://fed.brid.gy/ap/@mrpresidenttom@threads.net, but the follow still says 'requested' on Threads' side. And it doesn't look like @bsky.brid.gy@bsky.brid.gy followed my Threads account. I created a post (still on that 15 minute timer that Threads has), but I imagine since the follow didn't go through the post will never make it to bluesky.

**Comment by snarfed** [comment]
Love it! And yeah we're still debugging interop with them, they 500 on some of our activities, including `Accept`s and maybe still `Follow`s from the bot account. Stay tuned...

**Comment by snarfed** [comment]
This is largely working now! Still some bumps, but I'm comfortable enough to close this.

**Comment by Jay2645** [comment]
Question about this - is Threads still officially supported?

I remember this working at one point (and I see some posts do indeed bridge; e.g. https://bsky.app/profile/turningfortune.threads.net.ap.brid.gy), but https://fed.brid.gy/login doesn't have a login button for Threads, and the "Mastodon" button doesn't work.

I can make a new issue if that'd be better, but I wanted to double-check that it wasn't something intentional first.

(I'm trying to convince someone on Threads to join Bluesky and they don't want to duplicate their posts, but before I mention the bridge I just want to double-check that I'm not making a mistake)

**Comment by mackuba** [comment]
I think some small number of Threads accounts are showing up in my stats - I'm guessing you probably need to follow the Bridgy Mastodon account on Threads to enable it

**Comment by Jay2645** [comment]
Sorry, to clarify - it does look like bridging is still working; I guess my question is if it's intentional that https://fed.brid.gy/login doesn't have a Threads button/doesn't let you log in to Mastodon using a Threads account.

I didn't want to open a new issue if the answer was "Threads/Fediverse support is on life support so we don't actively maintain it." 😅

**Comment by snarfed** [comment]
@mackuba is right! We don't do much specific for Threads, but as far as we know it should still work, both directions, at least as much as Threads fediverse itself works. 🙄

**Comment by snarfed** [comment]
Ah! Sadly yes, we don't yet have Threads login on the web UI itself. Bridging via following the bot should still work though!

Threads web login has been fully built and ready for something like a year now, but we've been stuck in Meta API approval hell for most of that time, even with the Threads team themselves actively trying to help us and expedite. As far as we can tell, the Meta review team isn't against Bridgy Fed, it's just been impossibly bureaucratic and Kafka-esque, and so far it's defeated us. 🤷‍♂️

cc @anujahooja
